specification/programmableconnectivity/data-plane/readme.md

```diff
@@ -0,0 +1,56 @@
+# Azure Programmable Connectivity
+
+> see https://aka.ms/autorest
+
+This is the AutoRest configuration file for Azure Programmable Connectivity.
+
+## Getting Started
+
+To build the SDKs for My API, simply install AutoRest via `npm` (`npm install -g autorest`) and then run:
+
+> `autorest readme.md`
+
+To see additional help and options, run:
+
+> `autorest --help`
+
+For other options on installation see [Installing AutoRest](https://aka.ms/autorest/install) on the AutoRest github page.
+
+---
+
+## Configuration
+
+### Basic Information
+
+These are the global settings for the Azure Programmable Connectivity.
+
+```yaml
+openapi-type: data-plane
+tag: package-2024-02-09-preview
+title: AzureProgrammableConnectivity
+```
+
+### Tag: package-2024-02-09-preview
+
+These settings apply only when `--tag=package-2024-02-09-preview` is specified on the command line.
+
+```yaml $(tag) == 'package-2024-02-09-preview'
+input-file:
+  - Azure.ProgrammableConnectivity/preview/2024-02-09-preview/openapi.json
+suppressions:
+  - code: ValidResponseCodeRequired
```

Comment:
Had you tried adding this suppression inline in the TypeSpec to keep it local? https://azure.github.io/typespec-azure/docs/troubleshoot/suppresswarnings

Reply:
This is an openapi validation rather than a typespec validation, but I followed the instructions here to localise it: https://github.com/Azure/autorest/blob/main/docs/generate/suppress-warnings.md

I also didn't actually know we could suppress Typespec validation! https://eng.ms/docs/products/azure-developer-experience/design/specs-pr-guides/pr-suppressions suggests that this isn't possible so I didn't try. I've used that to suppress another error relating to returning an empty body in our 302 response.